src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1320,54 +1387,67 @@ private void ImportThrow()
             EmitTrapCall();
         }
 
-        private void ImportLoadField(int token, bool isStatic)
+        private LLVMValueRef GetInstanceFieldAddress(StackEntry objectEntry, FieldDesc field)
         {
-            if (isStatic)
+            var objectType = objectEntry.Type ?? field.OwningType;
+            LLVMValueRef typedObjectValue;
+            if (objectType.IsValueType && !objectType.IsPointer && objectEntry.Kind != StackValueKind.NativeInt && objectEntry.Kind != StackValueKind.ByRef)
             {
-                throw new NotImplementedException("static ldfld");
+                typedObjectValue = LLVM.BuildAlloca(_builder, GetLLVMTypeForTypeDesc(objectType), "objptr");
```

Comment:
Based on your other work, you should be able to use CastIfNecessary to get a pointer rather than doing a copy here. Additionally, I think this would end up returning the address of the copy on the stack, which could be invalidated.

Reply:
CastIfNecessary cant cast an Array to a Pointer and I see no way enable it to do so.

Reply:
Is the pattern where you hit this something like this?
```
ldloc someStruct
ldflda someField
```
I don't totally understand the way that interacts with alloca, but it might be ok if this only happens when a struct is loaded by value.

Reply:
its actually
```
ldloc someStruct
ldfld someField
```
I'm sure ldflda would also hit this, but my test case was just trying to test the value of TwoByteStr.second. it's somewhat unfortunate because in this case someStruct is originating from the Shadow Stack pointer, so it would be nice optimization at some point in the future to fuse this sort of pattern into a single GEP.

Reply:
Ok. I think copying is at least kind of right for loading a struct by value since you could get a sequence like:
```
ldloc someStruct // copy of the struct
ldloca someStruct // pointer to the original version of the struct
ldc 1
stfld someField // stores 1 in the original version of the struct
ldfld someField // should still load the original value from the copy of the struct
```
If you take care of the other minor comments, I think you'll be good to merge. Thanks for working through these issues.

Reply:
I think everything is covered now